neuraxle/base.py

```diff
@@ -1756,6 +1752,38 @@ def apply_method(self, method: Callable, *kargs, **kwargs) -> 'BaseStep':
         self.wrapped = self.wrapped.apply_method(method, *kargs, **kwargs)
         return self
 
+
+    def mutate(self, new_method="inverse_transform", method_to_assign_to="transform", warn=True) -> 'BaseStep':
+        """
+        Mutate self, and self.wrapped. Please refer to :func:`~neuraxle.base.BaseStep.mutate` for more information.
+
+        :param new_method: the method to replace transform with, if there is no pending ``will_mutate_to`` call.
+        :param method_to_assign_to: the method to which the new method will be assigned to, if there is no pending ``will_mutate_to`` call.
+        :param warn: (verbose) wheter or not to warn about the inexistence of the method.
+        :return: self, a copy of self, or even perhaps a new or different BaseStep object.
+        """
+        new_self = BaseStep.mutate(self, new_method, method_to_assign_to, warn)
+        self.wrapped = self.wrapped.mutate(new_method, method_to_assign_to, warn)
+        return new_self
```

Comment:
Should we also call like `BaseStep.mutate(self, ...)` before calling mutate on the wrapped ? I think we need to do that on self too and not only on wrapped. But good catch that it is done on the wrapped! I think we still need both. Do you agree?

Reply:
yes we should. and I think this is what I have done ? BaseStep.mutate followed by self.wrapped.mutate. what do you mean ? I think it's fine as it is now resolving